app/src/main/java/ch/sdp/vibester/activity/SearchUserActivity.kt

```diff
@@ -0,0 +1,86 @@
+package ch.sdp.vibester.activity
+
+import android.content.ContentValues
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.Log
+import android.view.Window
+import android.widget.EditText
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import ch.sdp.vibester.R
+import ch.sdp.vibester.database.Database
+import ch.sdp.vibester.profile.UserProfile
+import ch.sdp.vibester.profile.UserProfileAdapter
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.ValueEventListener
+
+/**
+ * Search for users based on their usernames.
+ */
+class SearchUserActivity : AppCompatActivity() {
+    private var userProfileAdapter: UserProfileAdapter? = null
+    private var users: MutableList<UserProfile>? = null
+    private var recyclerView: RecyclerView? = null
+    private var searchEditText: EditText? = null
+    private var dbRef: DatabaseReference = Database.get().getReference("users")
```

Comment:
this can be a "val" since dbRef will not change and should not be changed

Reply:
done!